membership_prorate_variable_period/tests/test_membership_prorate_variable_period.py

```diff
@@ -20,220 +21,110 @@ def setUp(self):
+        receivable_type = cls.env["account.account.type"].create(
             {
-                "partner_id": self.partner.id,
-                "date_invoice": fields.Date.today(),
-                "account_id": account,
+                "name": "Test receivable account",
+                "type": "receivable",
+                "internal_group": "income",
             }
         )
-        self.env["account.invoice.line"].create(
+        receivable = cls.env["account.account"].create(
             {
-                "account_id": account,
-                "product_id": self.product.id,
-                "price_unit": self.product.list_price,
-                "name": "Membership w/o prorate",
-                "invoice_id": invoice.id,
-                "quantity": 1.0,
+                "name": "Test receivable account",
+                "code": "TEST_RA",
+                "user_type_id": receivable_type.id,
+                "reconcile": True,
             }
         )
+        cls.partner = cls.env["res.partner"].create(
+            {
+                "name": "Test",
+                "property_account_receivable_id": receivable.id,
+            }
+        )
+
+    def create_invoice(self, invoice_date):
+        invoice_form = Form(
+            self.env["account.move"].with_context(default_move_type="out_invoice")
+        )
+        invoice_form.invoice_date = invoice_date
+        invoice_form.partner_id = self.partner
+        with invoice_form.invoice_line_ids.new() as line_form:
+            line_form.product_id = self.product
+        return invoice_form.save()
+
+    def test_create_invoice_membership_product_wo_prorate(self):
+        self.product.membership_prorate = False
+        invoice = self.create_invoice(date.today())
         self.assertAlmostEqual(invoice.invoice_line_ids[0].quantity, 1.0, 2)
         self.assertTrue(self.partner.member_lines)
         self.assertEqual(self.partner.member_lines[0].state, "waiting")
         self.assertEqual(self.partner.membership_state, "waiting")
 
     def test_create_invoice_membership_product_prorate_fixed(self):
         self.product.membership_type = "fixed"
-        self.product.membership_date_from = "2017-01-01"
-        self.product.membership_date_to = "2017-12-31"
-        account = self.partner.property_account_receivable_id.id
-        invoice = self.env["account.invoice"].create(
-            {
-                "partner_id": self.partner.id,
-                "date_invoice": "2017-04-01",
-                "account_id": account,
-            }
-        )
-        self.env["account.invoice.line"].create(
-            {
-                "account_id": account,
-                "product_id": self.product.id,
-                "price_unit": self.product.list_price,
-                "name": "Membership prorate fixed",
-                "invoice_id": invoice.id,
-                "quantity": 1.0,
-            }
-        )
+        self.product.membership_date_from = date(2017, 1, 1)
```

Comment:
These changes are not needed. Odoo accepts strings as well.

Reply:
Should I change it as it was before or should I keep the change?